alacritty_terminal/src/term/mod.rs

```diff
@@ -687,6 +687,87 @@ impl VisualBell {
     }
 }
 
+/// LineDamage describes the current damage of a line, represented as two
+/// column coordinates describing the edges of the damage. Damage is valid if
+/// left <= right.
+#[derive(Clone)]
+pub struct LineDamage {
+    pub left: Column,
+    pub right: Column,
+    pub line: Line,
+}
+
+impl LineDamage {
+    /// Create a new invalid LineDamage, which has left=columns+1 and right=0.
+    /// This ensures that later update with min/max of any valid column
+    /// automatically creates a valid line damage without needing any
+    /// conditionals.
+    fn new(line: Line, columns: Column) -> LineDamage {
+        LineDamage{
+            left: columns+1,
+            right: Column(0),
+            line,
+        }
+    }
+
+    #[inline(always)]
+    pub fn reset(&mut self, columns: Column) {
+        self.left = columns+1;
+        self.right = Column(0);
+    }
+}
+
+struct Damage {
+    line_damage: Vec<LineDamage>,
+    full_damage: bool,
+    columns: Column,
+}
+
+impl Damage {
+    #[inline(always)]
+    fn new(lines: Line, columns: Column) -> Damage {
+        let mut damage = Damage{
+            line_damage: Vec::with_capacity(lines.0),
+            full_damage: false,
+            columns: columns,
+        };
+        for line in 0..lines.0 {
+            damage.line_damage.push(LineDamage::new(Line(line), columns));
```

Comment:
I do understand your trick, but I think creating a proper new constructor and just passing reversed bounds would make much more sense.

Reply:
Inverted bounds is a LineDamage internal optimization, which appears illogical outside context.

I personally find it more sensible to contain that logic in LineDamage, minimizing the amount of places it needs to be explained.

Reply:
make sense, so lets just name it `placeholder` for now, since we want to add a useless line, which make meaningful later.